collect/collect.go

```diff
@@ -228,6 +245,76 @@ func (i *InMemCollector) checkAlloc() {
 	runtime.GC()
 }
 
+func (i *InMemCollector) newCheckAlloc() {
+	inMemConfig, err := i.Config.GetInMemCollectorCacheCapacity()
+
+	var mem runtime.MemStats
+	runtime.ReadMemStats(&mem)
+	i.Metrics.Gauge("memory_heap_allocation", int64(mem.Alloc))
+	if err != nil || inMemConfig.MaxAlloc == 0 || mem.Alloc < inMemConfig.MaxAlloc {
+		return
+	}
+
+	// Figure out what fraction of the total cache we should remove. We'd like it to be
+	// enough to get us below the max capacity, but not TOO much below.
+	// Because our impact numbers are only the data size, we shoot for 90% of max
+	// capacity and we should end up comfortably below that.
+	totalToRemove := mem.Alloc - inMemConfig.MaxAlloc //*9/10
+
+	// the size of the cache exceeds the user's intended allocation, so we're going to
+	// remove the traces from the cache that have had the most impact on allocation.
+	// To do this, we sort the traces by their CacheImpact value and then remove traces
+	// until the total size is less than 90% of the intended max.
+
+	existingCache, ok := i.cache.(*cache.DefaultInMemCache)
+	if !ok {
+		i.Logger.Error().WithField("alloc", mem.Alloc).Logf(
+			"total allocation exceeds limit, but unable to control cache",
+		)
+		return
+	}
+	allTraces := existingCache.GetAll()
+	timeout, err := i.Config.GetTraceTimeout()
+	if err != nil {
+		timeout = 60 * time.Second
```

Comment:
Is this refinery's default timeout?

Reply:
It is. This is only used in an error condition, which only happens if you specify an unparseable value for a timeout.